File: data/src/main/java/com/github/mrmitew/skeleton/data/http/OkHttpCache.java
package com.github.mrmitew.skeleton.data.http;

import android.content.Context;

import java.io.File;

import javax.inject.Inject;
import javax.inject.Singleton;

import okhttp3.Cache;

@Singleton
public class OkHttpCache {
    private Context mContext;
    private Cache mCache;

    @Inject
    public OkHttpCache(Context context) {
        mContext = context;
    }

    public Cache create() {
        mCache = null;
        try {
            mCache = new Cache(new File(mContext.getCacheDir(), "http-mCache"),
                    10 * 1024 * 1024); // 10 MB
        } catch (Exception e) {
            e.printStackTrace();
        }
        return mCache;
    }

    public Cache getCache() {
        return mCache;
    }
}

File: data/src/main/java/com/github/mrmitew/skeleton/data/http/OkHttpCache.kt
package com.github.mrmitew.skeleton.data.http

import android.content.Context
import okhttp3.Cache
import java.io.File
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class OkHttpCache @Inject constructor(context : Context){

    private val mContext : Context = context
    private var mCache: Cache? = null

    fun create() : Cache? {
         mCache = try {
            Cache(File(mContext.cacheDir, "http-mCache"),
                    10 * 1024 * 1024)
        } catch (e : Exception) {
            e.printStackTrace()
            return null
        }

        return mCache
    }

    fun getCache() : Cache? {return mCache}
}